roles/reproducer/tasks/push_code.yml

```diff
@@ -7,6 +7,22 @@
   delegate_to: controller-0
   delegate_facts: true
   block:
+    - name: Check if repository directories already exist
+      when: job_id is defined
+      ansible.builtin.stat:
+        path: "{{ item.project.src_dir }}"
+      register: repos_dir_stats
+      with_items: "{{ zuul['items'] }}"
+
+    - name: Ensure we are not in the job_id branch
+      when: repos_dir_stats is defined and item.1.stat.exists
```

Comment:
I'd rather see this syntax:
```YAML
- name: foobar
  when:
    - condition_1
    - condition_2
    - ....
```

Reply:
I'll change it since I have to rebase the PR because is out-of-date with the base branch